.github/workflows/goreleaser.yml

```diff
@@ -3,6 +3,9 @@ name: goreleaser
 on:
   workflow_dispatch:
 
+permissions:
```

Comment:
`goreleaser` will create release and upload assets. Does it need write permission?

Reply:
`goreleaser` seems to be using a separate GitHub token (being populated by the `GPR_TOKEN` GitHub secret):
https://github.com/fatedier/frp/blob/dev/.github/workflows/goreleaser.yml#L34

Hence the GitHub Actions generated token doesn't need any write permissions.